### DISCUSSION TITLE
How can I feed a top-level CUE value into an expression during export?

### DISCUSSION THREAD

**Discussion by cueckoo** [cue-lang/cue#1054] [Q&A]
_Originally opened by @seh in https://github.com/cuelang/cue/discussions/1054_

Suppose I have CUE file with the following content:
```cue
[
	{
		a: 1
	},
	{
		b: 2
	},
]
```
It's a top-level list, not a struct with any field names.

Suppose I'd like to export it as [a multi-document YAML stream](https://pkg.go.dev/cuelang.org/go/pkg/encoding/yaml#MarshalStream). Ideally I'd be able to refer to this top-level value in a command like this:
> _cue export list.cue --out=text --expression='yaml.MarshalStream($)'_

Here I used the identifier `$` in an appeal to Jsonnet, as some way to refer to the top-level value, but of course `$` is not understood by CUE.

I experimented with using the `--path` and `--with-context` flags, to no avail.

**Comment by myitcv** [comment]
@seh - thanks for your patience whilst we complete the migration (#1078)

Per our offline conversation, I'm not aware of a way of achieving this. Indeed in https://github.com/cue-lang/cue/issues/564#issuecomment-873388085 @mpvl and I discussed the possibility of using a bare `.` to refer to the current scope. 

So in this case that would look like:

```
cue export list.cue --out=text --expression='yaml.MarshalStream(.)'
```

How does that sit with you?

**Reply by seh** [reply]
That sounds like a good choice, assuming that there's no syntactic ambiguity there. There's plenty of precedent in other "path selection" languages.

**Reply by myitcv** [reply]
Great. I've marked this for discussion with @mpvl. I'll create a new issue if it turns out we don't repurpose #564.
